File: JavaScript/abtract-syntax-tree/convert.js
const fs = require('fs');
const { parse } = require('abstract-syntax-tree');


const code = `
function exampleFunction(x) {
    if (x > 10) {
        return x * 2;
    } else {
        return x + 2;
    }
}

class ExampleClass {
    constructor(value) {
        this.value = value;
    }

    display() {
        console.log(\`The value is \${this.value}\`);
    }
}

const exampleInstance = new ExampleClass(42);
exampleInstance.display();
`;

// Parse the code into an AST
const parsedAst = parse(code, { locations: true, range: true });

// Convert the AST to a formatted string
const formattedAst = JSON.stringify(parsedAst, null, 4);


fs.writeFile('result.json', formattedAst, (err) => {
    if (err) throw err;
    console.log('Formatted AST has been written to result.json');
});

File: Python/convert.py
import ast

code = """
def example_function(x):
    if x > 10:
        return x * 2
    else:
        return x + 2

class ExampleClass:
    def __init__(self, value):
        self.value = value

    def display(self):
        print(f"The value is {self.value}")

example_instance = ExampleClass(42)
example_instance.display()
"""

# Parse the code into an AST
parsed_ast = ast.parse(code)

# Convert the AST to a formatted string with indentation
ast_str = ast.dump(parsed_ast, indent=4)


with open('result.txt', 'w') as file:
    file.write(ast_str)

print("Formatted AST has been written to result.txt")

